core/src/main/java/org/jruby/RubyBasicObject.java

```diff
@@ -1718,27 +1718,35 @@ public static IRubyObject method_missing19(ThreadContext context, IRubyObject re
     public IRubyObject send(ThreadContext context, IRubyObject arg0, Block block) {
         String name = RubySymbol.objectToSymbolString(arg0);
 
-        return getMetaClass().finvoke(context, this, name, block);
+        StaticScope staticScope = context.getCurrentStaticScope();
+
+        return getMetaClass().finvokeWithRefinements(context, this, staticScope, name, block);
```

Comment:
Are these changes needed?  You changed finvoke to finvokeWithRefinements.

Reply:
finvoke will pass null for the scope when calling finvokeWithRefinements. This is the pattern they use in MRI... the non-refined logic calls the refined logic with a null scope.

Reply:
ok yeah I missed that